contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1672,3 +1694,95 @@ is_active_login(Oid role_oid)
 	return true;
 }
 
+/* 
+ * convertToUPN - This function is called to convert 
+ * domain\user to user@DOMAIN.
+ */
+char *
+convertToUPN(char* input)
+{
+	char *pos_slash = NULL;
+
+	if ((pos_slash = strchr(input, '\\')) != NULL)
```

Comment:
Before this, we should check whether the input is even of the format domain\user. Otherwise, throw an error. 
```
'foo' is not a valid Windows NT name. Give the complete name: <domain\username>.
```

Reply:
Done!